### ISSUE TITLE
Repository update does not consider options argument

### ISSUE THREAD

**Issue by rhogeranacleto** [typeorm/typeorm#3871]
**Issue type:**

[ ] question
[X] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[X] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[X] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

When calling method **update** from repository or manager passing a _SaveOptions_, it does not consider any of these options.
How did I notice, the following code snippet should not call my subscribers, but it does.

```ts
await getRepository(Entity).update(id, {
  name: 'updated name'
}, {
  listeners: false
});
```

Looking at the implementation:
https://github.com/typeorm/typeorm/blob/ba4f37f4fd61852c6371bd131047fcc760954204/src/entity-manager/EntityManager.ts#L498-L518

It just not uses the _options_ argument.
I think it should be used **or** just be removed from parameters :thinking: .

**Comment by rustamwin** [comment]
Yeah, in the next release will be removed

**Comment by rhogeranacleto** [comment]
@rustamwin Don't you think is a good idea to use it for disabling listeners?
Using a simple `.callListeners(options.listeners)`? I don't know if you agree, but I guess it's a good one :slightly_smiling_face:

**Comment by rustamwin** [comment]
@rhogeranacleto If you want disable listeners, what about using method `save` ? 
`getRepository(Entity).save(entity, {listeners: false})`

**Comment by rhogeranacleto** [comment]
@rustamwin Yeah, but think the following: I am executing something on a subscriber, and I don't want that _subscribers_ and _listeners_ be called again, I also want to execute a simple and direct update on the database, without pass over through process of _save method_ (it's quite complex).
I know that I can just create a query builder and do `.callListeners(false)` (is what I've done really), instead of using the _update_ method... :confused:

**Comment by pleerock** [comment]
@rustamwin I have reverted your PR. Please don't merge PRs until I review them. You should never merge breaking changes into master, and this is a breaking change. People may rely on listeners for insert/update operation and changes break their exist functionality.

Regarding to this particular functionality, I'm not yet sure, I'm starting to think about deprecation of insert and update methods because people overuse them and then face with different problems and limitations.

But for now advice is to use query builder and disable listeners.

**Comment by chillinski** [comment]
Is there a way to just update one value on an entity without materializing the complete entity, changing a value of one field and saving it back to the database? We use PostGreSQL as database layer and save does not work in some cases as unique constraints are violated.
